src/electionguard/election_polynomial.py

```diff
@@ -19,55 +22,68 @@
 )
 from .schnorr import make_schnorr_proof, SchnorrProof
 
+# TODO: rename to just polynomial and improve the documentation with some links to wikipedia
 
+# TODO: do not use lists here as they are susceptible to order-based attacks
 class ElectionPolynomial(NamedTuple):
     """
-    ElectionPolynomial is a polynomial defined by coefficients. The 0 coefficient is used for a secret key which can 
+    ElectionPolynomial is a polynomial defined by coefficients. 
+    The 0-index coefficient is used for a secret key which can 
     be discovered by a quorum of n guardians corresponding to n coefficients.
     """
 
     coefficients: List[ElementModQ]
+    """
+    The secret coefficients `a_ij`
+    """
+
     coefficient_commitments: List[ElementModP]
+    """
+    The public keys `K_ij`generated from secret coefficients
+    """
+
     coefficient_proofs: List[SchnorrProof]
+    """
+    A proof of posession of the private key for each secret coefficient
+    """
 
 
-def generate_polynomial(number_of_coefficients: int) -> ElectionPolynomial:
+def generate_polynomial(
+    number_of_coefficients: int, nonce: ElementModQ = None
+) -> ElectionPolynomial:
     """
     Generates a polynomial for sharing election keys
 
     :param number_of_coefficients: Number of coefficients of polynomial
+    :param nonce: an optional nonce parameter that may be provided (useful for testing)
     :return: Polynomial used to share election keys
     """
     coefficients: List[ElementModQ] = []
     commitments: List[ElementModP] = []
     proofs: List[SchnorrProof] = []
-    polynomial = ElectionPolynomial(coefficients, commitments, proofs)
-
-    assert number_of_coefficients > 0, "must have a positive number of coefficients"
-
-    # TODO: you might want to have an optional argument -- a nonce -- from which you derive all
-    #   the coefficients, rather than making them all be random here. That would make this function
-    #   deterministic and thus more testable. Of course, when used "for real", you'd make sure that
-    #   the input nonce was truly random.
 
     for i in range(number_of_coefficients):
-        coefficient = rand_q()
+        # Note: the nonce value is not safe.  it is designed for testing only.
+        # this method should be called without the nonce in production.
+        coefficient = (
+            int_to_q_unchecked(nonce.to_int() + i) if nonce is not None else rand_q()
```

Comment:
I want this:
`nonce.to_q() + i if nonce is not None else rand_q()`

I feel like that would read better. Understandable it may not be a move we can make.